### ISSUE TITLE
DOODS just stop sometimes

### ISSUE THREAD

**Issue by r3dsub** [snowzach/hassio-addons#40]
Here at 4:31 logging just stop and I rebooted at 09:09 manually. 
It also downloads requests every time again when I Start it. Is it normal behaviour? 

2025-02-23 04:31:55,798 - uvicorn.access - INFO - 192.168.178.72:39790 - "POST /detect HTTP/1.1" 200
2025-02-23 04:31:57,809 - uvicorn.access - INFO - 192.168.178.72:39804 - "POST /detect HTTP/1.1" 200
2025-02-23 04:31:57,808 - doods.doods - INFO - DetectResponse(id='', image=None, detections=[], error=None, duration=17.271041055209935)
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
2025-02-23 09:09:24,742 - doods.doods - INFO - Registered detector type:tflite name:default
2025-02-23 09:09:25,792 - doods.doods - INFO - Registered detector type:tensorflow name:tensorflow
/usr/local/lib/python3.11/dist-packages/torch/hub.py:293: UserWarning: You are about to download and run code from an untrusted repository. In a future release, this won't be allowed. To add the repository to your trusted list, change the command to {calling_fn}(..., trust_repo=False) and a command prompt will appear asking for an explicit confirmation of trust, or load(..., trust_repo=True), which will assume that the prompt is to be answered with 'yes'. You can also use load(..., trust_repo='check') which will only prompt for confirmation if the repo is not already trusted. This will eventually be the default behaviour
  warnings.warn(
Downloading: "https://github.com/ultralytics/yolov5/zipball/master" to /root/.cache/torch/hub/master.zip
[31m[1mrequirements:[0m Ultralytics requirement ['requests>=2.32.2'] not found, attempting AutoUpdate...
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip is available: 23.2.1 -> 25.0.1
[notice] To update, run: python3 -m pip install --upgrade pip
Collecting requests>=2.32.2
  Obtaining dependency information for requests>=2.32.2 from https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl.metadata
  Downloading requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.32.2) (3.2.0)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests>=2.32.2) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests>=2.32.2) (2.0.5)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests>=2.32.2) (2023.7.22)
Downloading requests-2.32.3-py3-none-any.whl (64 kB)
YOLOv5 🚀 2025-2-23 Python-3.11.0rc1 torch-2.3.1+cpu CPU

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 1.1 MB/s eta 0:00:00
Installing collected packages: requests
  Attempting uninstall: requests
    Found existing installation: requests 2.31.0
    Uninstalling requests-2.31.0:
      Successfully uninstalled requests-2.31.0
Successfully installed requests-2.32.3

[31m[1mrequirements:[0m AutoUpdate success ✅ 1.5s, installed 1 package: ['requests>=2.32.2']
[31m[1mrequirements:[0m ⚠️ [1mRestart runtime or rerun command for updates to take effect[0m

Downloading https://github.com/ultralytics/yolov5/releases/download/v7.0/yolov5s.pt to yolov5s.pt...

  0%|          | 0.00/14.1M [00:00<?, ?B/s]
  4%|▎         | 512k/14.1M [00:00<00:02, 4.94MB/s]
 20%|██        | 2.88M/14.1M [00:00<00:00, 16.1MB/s]
 42%|████▏     | 5.88M/14.1M [00:00<00:00, 22.8MB/s]
 69%|██████▉   | 9.75M/14.1M [00:00<00:00, 29.6MB/s]
 92%|█████████▏| 13.0M/14.1M [00:00<00:00, 31.1MB/s]
100%|██████████| 14.1M/14.1M [00:00<00:00, 27.7MB/s]

Fusing layers... 
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients, 16.4 GFLOPs
Adding AutoShape... 
2025-02-23 09:09:31,048 - doods.doods - INFO - Registered detector type:pytorch name:pytorch
2025-02-23 09:09:31,082 - uvicorn.error - INFO - Started server process [7]
2025-02-23 09:09:31,082 - uvicorn.error - INFO - Waiting for application startup.
2025-02-23 09:09:31,083 - uvicorn.error - INFO - Application startup complete.
2025-02-23 09:09:31,083 - uvicorn.error - INFO - Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
2025-02-23 12:08:31,337 - uvicorn.access - INFO - 192.168.178.78:35278 - "GET /detectors HTTP/1.1" 200
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
2025-02-24 02:28:1

**Comment by r3dsub** [comment]
Seems it was due to Lack of RAM
